docs/utxo-hd/v0.1/for-public.md

```diff
@@ -0,0 +1,166 @@
+on rare occasions.
+
+## Impact on a passive running node
+
+When running a non block producing node that is caught up with the network, the
+current throughtput of blocks is roughly 1 every 20 seconds, so the node has
+plenty of time to perform I/O to fetch inputs and apply the blocks. In this
+case, a pre-UTxO-HD node doesn't run any faster than the UTxO-HD node as both
+nodes can easily process one block per twenty seconds, as opposed to syncing and
+replaying which are operations whose aim is to process a big pool of data.
+
+## Impact on a block producer node
+
+There is a slight difference when the node is a block producer, as those also
+manage a mempool to validate transactions. These transactions also have to go
+through the same workflow as block application and therefore need to perform I/O
+to fetch arbitrary inputs from disk. There is in principle some slowdown in this
+case. However, we put effort into reducing this impact and by maintaining cached
+read views of the disk data. This ensures that the node can function properly. A
+small impact is expected, but in general (given that blocks are produced every
+20 seconds) there is enough time for the node to fetch data from the disk such
+that the impact should not be very noticeable.
+
+# Future protocols of Cardano
+
+With the implementation of [Ouroboros
+Leios](https://iohk.io/en/research/library/papers/ouroboros-leios-design-goals-and-concepts/),
+blocks will be produced faster than they are currently, so there arises a need
+to use more advanced solutions to saturate the throughtput of SSDs and maximize
+parallelism when operating on the changes to the UTxO set (or other data
+structures).
+
+For this goal, there is ongoing development to produce an [LSM
+tree](https://en.wikipedia.org/wiki/Log-structured_merge-tree) backend
+implementation that will provide us with faster access to the data. This is
+mainly achieved by tailoring the implementation to the needs of the node. There
+is also the prospect of using pipelining of blocks and fetch of inputs to speed
+up the operations.
+
+# Querying of the node
+
+*Note: this section is still under discussion, so take it with a grain of salt,
+this is just my own wording (@jasagredo).*
+
+There is no need for the node to provide fast access to the ledger state data,
+at least not for its own goals, but it is true that it comes in handy. As much
+as possible, the node aims to provide this access, but due to the way the data
+is processed, there is a particular query that has become much more expensive
+than before: `GetUTxOByAddress`.
+
+Before UTxO-HD, this query needed to traverse the whole UTxO set, filtering for
+the UTxOs owned by a particular addres. Now this has to be done by querying the
+disk, which makes this quite costly.
+
+This is an undesirable effect of UTxO-HD that will result in a need of an
+external indexer service that keeps a table of UTxOs indexed by addresses. This
+service still doesn't exist.
```

Comment:
Perhaps we can link to https://github.com/input-output-hk/cardano-node/issues/4678

Reply:
Done